File: InterconnectFrontend/src/models/interfaces/BaseEntity.ts
/**
 * Base interface for all entities in the simulation with position information.
 */
interface BaseEntity {
    /** Unique identifier for the entity */
    id: number;
    /** X coordinate position in the simulation stage */
    x: number;
    /** Y coordinate position in the simulation stage */
    y: number;
}

export default BaseEntity;

File: InterconnectBackend/Models/BaseEntity.cs
namespace Models
{
    /// <summary>
    /// Base class representing an entity with a position on the board.
    /// </summary>
    public class BaseEntity
    {
        /// <summary>
        /// Entity identifier.
        /// </summary>
        public required int Id { get; set; }
        
        /// <summary>
        /// X coordinate of the entity position on the board.
        /// </summary>
        public required int X { get; set; }
        
        /// <summary>
        /// Y coordinate of the entity position on the board.
        /// </summary>
        public required int Y { get; set; }
    }
}
